content/docs/components/serving/bentoml.md

```diff
@@ -0,0 +1,234 @@
+    * `kubectl` install instruction: https://kubernetes.io/docs/tasks/tools/install-kubectl/
+2. Docker and Docker Hub installed and configured in your local machine.
+    * Docker install instruction: https://docs.docker.com/get-docker/
+3. Python 3.6 or above and required PyPi packages: `bentoml`, `scikit-learn`
+    * ```pip install bentoml scikit-learn```
+
+BentoML is an open-source framework for high-performance ML model serving, which supports
+all major machine learning frameworks including Keras, Tensorflow, PyTorch, Fast.ai,
+XGBoost and etc.
+
+### Build an iris classifier model server with BentoML
+
+Save the following code to a new file named `iris_classifier.py`:
+
+```python
+from bentoml import env, artifacts, api, BentoService
+from bentoml.handlers import DataframeHandler
+from bentoml.artifact import SklearnModelArtifact
+
+@env(auto_pip_dependencies=True)
+@artifacts([SklearnModelArtifact('model')])
+class IrisClassifier(BentoService):
+
+    @api(DataframeHandler)
+    def predict(self, df):
+        return self.artifacts.model.predict(df)
+```
+
+This code defines a model API server that requires a `scikit-learn` model, and asks
+BentoML to figure out the required PyPI packages automatically. It also defined an
+API, which is the entry point for accessing this prediction service. And the API is
+expecting a `pandas.DataFrame` object as its input data.
+
+Run the following code to create a BentoService SavedBundle with the iris classification
+model. A BentoService SavedBundle is a versioned file archive ready for production
+deployment. The archive contains the model service defined above, python code
+dependencies, PyPi dependencies, and the trained iris classification model:
+
+```python
+from sklearn import svm
+from sklearn import datasets
+
+from iris_classifier import IrisClassifier
+
+if __name__ == "__main__":
+    # Load training data
+    iris = datasets.load_iris()
+    X, y = iris.data, iris.target
+
+    # Model Training
+    clf = svm.SVC(gamma='scale')
+    clf.fit(X, y)
+
+    # Create a iris classifier service instance
+    iris_classifier_service = IrisClassifier()
+
+    # Pack the newly trained model artifact
+    iris_classifier_service.pack('model', clf)
+
+    # Save the prediction service to disk for model serving
+    saved_path = iris_classifier_service.save()
+```
+
+Use BentoML CLI to start a local API model server:
+
+```shell
+bentoml serve IrisClassifier:latest
+```
+
+BentoML automatically processes the incoming data into the required data format defined in
+the API. For the iris classifier BentoService defined above, incoming data transforms
+into a `pandas.DataFrame`.
+
+Use `curl` request in another terminal to get the prediction result:
+
+```bash
+curl -i \
+  --header "Content-Type: application/json" \
+  --request POST \
+  --data '[[5.1, 3.5, 1.4, 0.2]]' \
+  localhost:5000/predict
+```
+
+Find the file directory of the SavedBundle with `bentoml get` command, which is
```

Comment:
... with `bentoml get` command. The SavedBundle directory is structured as a ....